### PR TITLE
[kubernetes] Allow MySQL instance to be external

### PR DESCRIPTION
**Pull request by dd-willgan**
#### Issue(s)

- https://github.com/startreedata/thirdeye/issues/977

#### Description

- Use `condition` on the MySQL subchart so that local MySQL doesn't have to be deployed
- Add `.Values.mysql.mysqlUrl` so that external URL can be provided
- Updated configmaps so that `mysqlUrl` is used if provided

#### Screenshots

N/A

#### How to test

From `kubernetes/helm/startree-thirdeye`, I ran the following with different values of `mysql.enabled` and `mysql.mysqlUrl`. Saw working.

```
helm install --dry-run --debug thirdeye . > dry-run.txt
```

### CODE REVIEW

**Review by mmantri2022 — COMMENTED** [review]
Looks good to me. As long as we have ensured no security risk involved.

**Review by mmantri2022 — APPROVED** [review]
looks good to me as long as there is no security issues with this and right measure and password protection is taken care.

File: kubernetes/helm/startree-thirdeye/templates/coordinator/configmap.yaml [hunk 1]
```diff
@@ -59,8 +59,12 @@ data:
 {{ toYaml .Values.auth | indent 6 }}
 
     database:
-      # Assuming a local MySQL server running on the default port 3306
+      # Defaults to k8s cluster local MySQL server running on port 3306, but uses URL if provided
+      {{ if .Values.mysql.mysqlUrl }}
```
Comment (shounakmk219) [inline kubernetes/helm/startree-thirdeye/templates/coordinator/configmap.yaml:63]: The condition should be on `.Values.mysql.enabled`. That will force user to provide the `mysqlUrl` when internal mysql is disabled. Also avoid the situation of thirdeye connecting to external mysql as well as spinning its own instance if `mysql.mysqlUrl` is also set and `mysql.enabled` is also set to `true`. Let me know your thoughts.

File: kubernetes/helm/startree-thirdeye/templates/scheduler/scheduler-config.yaml [hunk 2]
```diff
@@ -83,8 +83,12 @@ data:
     #      ttl: 60000
 
     database:
-      # Assuming a local MySQL server running on the default port 3306
+      # Defaults to k8s cluster local MySQL server running on port 3306, but uses URL if provided
+      {{ if .Values.mysql.mysqlUrl }}
```
Comment (shounakmk219) [inline kubernetes/helm/startree-thirdeye/templates/scheduler/scheduler-config.yaml:87]: Same comment as above

File: kubernetes/helm/startree-thirdeye/templates/worker/worker-config.yaml [hunk 3]
```diff
@@ -83,8 +83,12 @@ data:
     #      ttl: 60000
 
     database:
-      # Assuming a local MySQL server running on the default port 3306
+      # Defaults to k8s cluster local MySQL server running on port 3306, but uses URL if provided
+      {{ if .Values.mysql.mysqlUrl }}
```
Comment (shounakmk219) [inline kubernetes/helm/startree-thirdeye/templates/worker/worker-config.yaml:87]: Same comment as above

**Review by shounakmk219 — CHANGES_REQUESTED** [review]
Nice addition! just few comments. 
You may also want to make changes at the [backup job](https://github.com/startreedata/thirdeye/blob/0d1a8011bbeb580ff5c7fdc86eccfa7344a06b70/kubernetes/helm/startree-thirdeye/templates/mysql/mysql-backup-cronjob.yaml#L51) so that it points to the right mysql instance.

**Review by shounakmk219 — APPROVED** [review]

**Pull request merged**